muse_payment_adaptor/api_interface/muse_api_client.py

```diff
@@ -0,0 +1,158 @@
+import json
+import logging
+from abc import ABC
+from typing import List, Tuple, Union
+
+from pydantic import BaseModel
+
+from core.datetimes.ad_datetime import datetime
+from invoice.models import Bill, BillItem
+from muse_payment_adaptor.api_interface.dataclasses import BulkyPaymentSubmissionRequest, \
+    BulkyPaymentSubmissionRequestMessage, BulkyPaymentMessageHeader, BulkyPaymentSummary, BulkyPaymentPayListElement
+from muse_payment_adaptor.api_interface.muse_http_client import MuseAdaptorHttpClient
+from muse_payment_adaptor.api_interface.signature import create_signature_b64_string
+from muse_payment_adaptor.apps import MusePaymentAdaptorConfig
+from muse_payment_adaptor.models import PaymentRequest, PaymentRequestDetails
+
+logger = logging.getLogger(__name__)
+_default_http_client = MuseAdaptorHttpClient(MusePaymentAdaptorConfig.muse_base_uri)
+
+
+# region ABC
+
+class MuseApiClient(ABC):
+    def __init__(self, api_http_client: MuseAdaptorHttpClient = _default_http_client):
+        self.api_http_client = api_http_client
+
+    @property
+    def _api_endpoint(self) -> str:
+        """
+        This property specifies the exact endpoint to be used by the client. The endpoint will be joined with root path
+        from the MuseAdaptorHttpClient.
+        """
+        raise NotImplementedError("`api_endpoint` not implemented")
+
+    def _submit(self, model: BaseModel) -> str:
+        response = self.api_http_client.send_post_request(self._api_endpoint, model.json())
+        if response.ok:
+            return response.content.decode('utf-8')
+        else:
+            logger.error(f"Muse request failed, HTTP code {response.status_code}")
```

Comment:
Do we want to specify any other action in here? Just logging error seems to be insufficient. Also besides status we could log response. 
When it comes to format of log message it is recommended to use % notation instead of f strings ([link](https://docs.python.org/3/howto/logging-cookbook.html#use-of-alternative-formatting-styles)).

Reply:
The submission is unrecoverable at this point so i don't think there is anything to do beside logging

Reply:
I added additional log for error content